src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,21 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 42;
+        result = result * 32 * width.hashCode();
+        result = result * 32 * length.hashCode();
+        result = result * 32 * color.hashCode();
```

Comment:
check for null before calling "field.hashcode()", cause you can get nullPointer in some cases